src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -629,15 +627,26 @@ private StudentEnrollDetails enrollStudent(StudentAttributes validStudentAttribu
     }
 
     private List<String> getInvalidityInfoInDuplicatedEmail(String email,
-            List<String> studentEmailList, String[] linesArray) {
+            List<StudentAttributes> studentList, String[] linesArray) {
         List<String> info = new ArrayList<>();
-        info.add("Same email address as the student in line \"" + linesArray[studentEmailList.indexOf(email) + 1] + "\"");
+        int index;
+        for (index = 0; index < studentList.size(); index++) {
+            if (studentList.get(index).email.equalsIgnoreCase(email)) {
+                break;
+            }
+        }
+        info.add("Same email address as the student in line \"" + linesArray[index + 1] + "\"");
         return info;
```

Comment:
do we need the `break` at line 635? Otherwise we can shift lines 638 & 639 there instead.